src/day8.scala

```diff
@@ -0,0 +1,136 @@
+// using scala 3.0.2
+
+package day8
+
+import scala.util.Using
+import scala.util.chaining.*
+import scala.io.Source
+
+import Segment.*
+import Digit.*
+
+@main def part1: Unit =
+  println(s"The solution is ${part1(readInput())}")
+
+@main def part2: Unit =
+  println(s"The solution is ${part2(readInput())}")
+
+def readInput(): String =
+  Using.resource(Source.fromFile("input/day8"))(_.mkString)
+
+enum Segment:
+  case A, B, C, D, E, F, G
+
+  val char = toString.head.toLower
+
+object Segment:
+  type Segments = Seq[Segment]
+
+  val fromChar: Map[Char, Segment] = values.map(s => s.char -> s).toMap
+
+  def parseSeq(s: String): Segments =
+    s.map(fromChar)
```

Comment:
Why is this called parse*Seq*? It doesn't seem to be related to `Seq` at all. Consider using a name that reflects the semantics more than the types involved. (The types are there to talk about the types.)

Reply:
changing to `parseSegments`